boards/PDM/Src/App/App_SetPeriodicCanSignals.c

```diff
@@ -0,0 +1,116 @@
+#include "App_SetPeriodicCanSignals.h"
+
+// Even if the value is out-of-range, we still use it to set the CAN TX signal
+static void App_SetPeriodicCanSignals_InRangeCheck(
+    struct PdmCanTxInterface *const  can_tx,
+    const struct InRangeCheck *const in_range_check,
+    void (*const can_signal_setter)(struct PdmCanTxInterface *, float),
+    void (*const underflow_setter)(struct PdmCanTxInterface *, uint8_t),
+    void (*const overflow_setter)(struct PdmCanTxInterface *, uint8_t))
+{
+    float                    value;
+    enum InRangeCheck_Status status =
+        App_InRangeCheck_GetValue(in_range_check, &value);
+
+    switch (status)
+    {
+        case VALUE_IN_RANGE:
+        {
+            underflow_setter(can_tx, false);
+            overflow_setter(can_tx, false);
+        }
+        break;
+        case VALUE_UNDERFLOW:
+        {
+            underflow_setter(can_tx, true);
+        }
+        break;
+        case VALUE_OVERFLOW:
+        {
+            overflow_setter(can_tx, true);
+        }
+        break;
+    }
+
+    can_signal_setter(can_tx, value);
+}
```

Comment:
As a thought, this whole function could be a macro `APP_CREATE_PERIODIC_CAN_SIGNALS_IN_RANGE_CHECK(CAN_INTERFACE_TYPE)` so that we can share it between boards.

Reply:
Oh yeah let's do that before I merge so @pasitlaothamatas can use it for his FSM wheel speed PR as well. How do we want to design the macro to paste the `BoardName` in to `CanTxInterface` though? Or we could even just leave it as a void pointer so we can leave it as a function instead of a macro?

Reply:
I am fairly strongly against tossing away typing with void pointers in basically all cases. I think pasting the `BoardName` is fine, ie:
```
APP_CREATE_PERIODIC_CAN_SIGNALS_IN_RANGE_CHECK(CAN_INTERFACE_TYPE) \
static void App_SetPeriodicCanSignals_InRangeCheck( \
    struct CAN_INTERFACE_TYPE *const  can_tx, \
    const struct InRangeCheck *const in_range_check, \
    void (*const can_signal_setter)(struct CAN_INTERFACE_TYPE *, float), \
    void (*const underflow_setter)(struct CAN_INTERFACE_TYPE *, uint8_t), \
    void (*const overflow_setter)(struct CAN_INTERFACE_TYPE *, uint8_t)) \
{ \
    float                    value; \
    enum InRangeCheck_Status status = \
        App_InRangeCheck_GetValue(in_range_check, &value); \
 \
    switch (status) \
    { \
        case VALUE_IN_RANGE: \
        { \
            underflow_setter(can_tx, false); \
            overflow_setter(can_tx, false); \
        } \
        break; \
        case VALUE_UNDERFLOW: \
        { \
            underflow_setter(can_tx, true); \
        } \
        break; \
        case VALUE_OVERFLOW: \
        { \
            overflow_setter(can_tx, true); \
        } \
        break; \
    } \
\
    can_signal_setter(can_tx, value); \
}

```